js/components/videoQualityPanel.js

```diff
@@ -24,53 +27,164 @@ var VideoQualityPanel = React.createClass({
     this.setState({
       selected: selectedBitrateId
     });
-    this.props.togglePopoverAction();
+    this.props.closeAction({
+      restoreToggleButtonFocus: true
+    });
+  },
+
+  /**
+   * Keydown event handler. Implements arrow key navigation for menu items.
+   * TODO:
+   * Export this logic to a generic higher order component so that this is reusable
+   * by other menu components.
+   * @private
+   * @param {event} event The keyboard event object.
+   */
+  handleVideoQualityKeyDown: function(event) {
+    if (!event.target || !event.target.hasAttribute(CONSTANTS.KEYBD_FOCUS_ID_ATTR)) {
+      return;
+    }
+    switch (event.key) {
+      case CONSTANTS.KEY_VALUES.ARROW_UP:
+      case CONSTANTS.KEY_VALUES.ARROW_LEFT:
+        event.preventDefault();
+        this.focusOnMenuItemSibling(event.target, false);
+        break;
+      case CONSTANTS.KEY_VALUES.ARROW_DOWN:
+      case CONSTANTS.KEY_VALUES.ARROW_RIGHT:
+        event.preventDefault();
+        this.focusOnMenuItemSibling(event.target, true);
+        break;
+      default:
+        break;
+    }
+  },
+
+  /**
+   * Finds the previous or next sibling of the given menu item and gives it focus.
+   * @private
+   * @param {Element} menuItem The menuItem element whose sibling we want to focus on.
+   * @param {Boolean} useNextSibling Choses the sibling next to menuItem when true and the previous one when false.
+   */
+  focusOnMenuItemSibling: function(menuItem, useNextSibling) {
+    var menuItemsList = [];
+    if (this.menuDomElement) {
+      menuItemsList = this.menuDomElement.querySelectorAll('[' + CONSTANTS.KEYBD_FOCUS_ID_ATTR + ']');
+    }
+    if (!menuItemsList.length) {
+      return;
+    }
+    // Since these elements aren't actually next to each other in the DOM, their position
+    // relative to one another is implied from their tab order, which should be the same as
+    // the one returned by querySelectorAll as long as tabindex is set to 0 (which should be the case).
```

Comment:
How possible would it be for this to cause un-intuitive UI? Like pressing left would move right, etc?

Reply:
It's possible but you would have to arrange the elements in a weird way in the html or with css. This will navigate in the same order as it does when tabbing, so if the tab ordering is un-intuitive, the arrow keys will be awkward as well.